### PR TITLE
Remove react-resizable

### PR DESCRIPTION
**Pull request by flenter**
Requestor page (routes pannel & request panel) now use the `react-resizable-panel` library

### CODE REVIEW

**Comment by brettimus** [comment]
needs a conflict resolution with recent changes ((from #153 i think)) 

we'll probably merge late today or monday since we're going to cut releases off main soon

**Comment by flenter** [comment]
There was a minor conflict with the changes from https://github.com/fiberplane/fpx/pull/161.  But i've resolved them

File: frontend/src/hooks/useIsSmScreen.ts [hunk 1]
```diff
@@ -6,5 +6,5 @@ import { useMedia } from "@fiberplane/hooks";
  * IMPROVE - Use css variable instead of hardcoded 640px?
  */
 export function useIsSmScreen() {
-  return useMedia("(min-width: 640px)");
+  return useMedia("(max-width: 640px)");
```
Comment (brettimus) [inline frontend/src/hooks/useIsSmScreen.ts:9]: so i named this hook poorly, but the intention was: "return true if it matches the `sm` breakpoint", to have similar logic to the `sm:` modifier in tailwind classes. 

i will look for potential breakage elsewhere

File: frontend/src/pages/RequestorPage/RequestorPage.tsx [hunk 2]
```diff
@@ -209,146 +223,220 @@ export const RequestorPage = () => {
...
+              <RoutesPanel
+                routes={routes}
+                selectedRoute={selectedRoute}
+                handleRouteClick={handleSelectRoute}
+                history={history}
+                loadHistoricalRequest={loadHistoricalRequest}
+                removeServiceUrlFromPath={removeServiceUrlFromPath}
+              />
+            </ResizablePanel>
+            <ResizableHandle
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestorPage.tsx:344]: image

look out: creates a weird double border

**Comment by brettimus** [comment]
pleasantly surprised that the layout didn't break too much hahaha.

on smaller screens for, e.g., POST requests, i often found that the out-of-the-box width would make it so the `Body` tab would overflow, and that would (i think) push the `Erase` icon for clearing params out of view as well:

image

also happened on larger screens quite often UNTIL i manually resized the panels. then the layout is persisted

File: frontend/src/hooks/useIsSmScreen.ts (hunk 1 above)
Comment (brettimus) [reply frontend/src/hooks/useIsSmScreen.ts:9]: okay. requestorpage is the only component that uses this. we're good to go

File: frontend/src/pages/RequestorPage/RequestorPage.tsx [hunk 3]
```diff
@@ -209,146 +223,220 @@ export const RequestorPage = () => {
...
+  );
+
   return (
     <div
       className={cn(
         // It's critical the parent has a fixed height for our grid layout to work
         "h-[calc(100vh-64px)]",
         // We want to `grid` all the things
-        "grid",
+        // "grid",
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestorPage.tsx]:
```suggestion
```

File: frontend/src/pages/RequestorPage/RequestorPage.tsx [hunk 4]
```diff
@@ -209,146 +223,220 @@ export const RequestorPage = () => {
...
+        "flex-col",
         "gap-2",
         "py-4 px-2",
         "sm:px-4 sm:py-3",
         // Define row templates up until the `lg` breakpoint
-        "max-lg:grid-rows-[auto_1fr]",
+        // "max-lg:grid-rows-[auto_1fr]",
         // Define column templates for the `lg` breakpoint
-        "lg:grid-cols-[auto_1fr]",
+        // "lg:grid-cols-[auto_1fr]",
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestorPage.tsx]: can delete all this jazz

File: frontend/src/pages/RequestorPage/RequestorPage.tsx [hunk 5]
```diff
@@ -209,146 +223,220 @@ export const RequestorPage = () => {
...
-        </div>
-      </div>
+          </>
+        )}
+        <ResizablePanel>
+          <div
+            className={cn(
+              "flex",
+              "flex-col",
+              // This is a custom css class that uses the famed `auto minmax(0, 1fr)` trick
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestorPage.tsx]: do we need this `fpx-requestor-grid-rows` class anymore now that it's flex?

File: frontend/src/pages/RequestorPage/RoutesPanel.tsx [hunk 6]
```diff
@@ -36,8 +33,8 @@ export function RoutesPanel({
   loadHistoricalRequest,
   removeServiceUrlFromPath,
 }: RoutesPanelProps) {
-  const { width, handleResize } = useResizableWidth(320);
-  const styleWidth = useStyleWidth(width);
+  // const { width, handleResize } = useResizableWidth(320);
+  // const styleWidth = useStyleWidth(width);
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RoutesPanel.tsx]:
```suggestion
```

**Review by brettimus — APPROVED** [review]

File: frontend/src/pages/RequestorPage/RequestorPage.tsx (hunk 5 above)
Comment (brettimus) [reply frontend/src/pages/RequestorPage/RequestorPage.tsx]: answer: we do not (just checked locally)

File: frontend/src/pages/RequestorPage/RequestorPage.tsx [hunk 7]
```diff
@@ -209,146 +223,220 @@ export const RequestorPage = () => {
...
+              "fpx-requestor-grid-rows",
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestorPage.tsx]:
```suggestion
```

**Pull request merged**